src/main/java/seedu/savenus/model/savings/Savings.java

```diff
@@ -0,0 +1,71 @@
+package seedu.savenus.model.savings;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.savenus.commons.util.AppUtil.checkArgument;
+
+/**
+ * Acts as the amount of money to be added into the {@code SavingsAccount} of the user
+ * and also the amount of money to deduct from the {@code Wallet} of the user.
+ */
+public class Savings {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Please provide a savings amount with 0 or 2 decimal places.\n"
+            + "For example: 1.50 or 200";
+
+    public static final String VALIDATION_REGEX = "(0|(0(\\.\\d{2,2}))|[1-9]+(\\d*(\\.\\d{2,2})?))";
+
+    private final String savingsAmount;
+
+    public Savings(String savings) {
+        requireNonNull(savings);
+        checkArgument(isValidSaving(savings), MESSAGE_CONSTRAINTS);
+        savingsAmount = convert(savings);
+    }
+
+    /**
+     * Returns true if a given string is a valid saving representation.
+     */
+    public static boolean isValidSaving(String saving) {
+        if (saving.matches(VALIDATION_REGEX)) {
+            if (Double.parseDouble(saving) > 5000.00) {
+                return false;
```

Comment:
Does this mean that users cannot save more than 5000 dollars at once?

Reply:
Ah didn't notice that I set this limit. Will work on it now.